src/ripple/app/tx/impl/Change.cpp

```diff
@@ -221,4 +235,127 @@ Change::applyFee()
     return tesSUCCESS;
 }
 
+TER
+Change::applyUNLModify()
+{
+    if (view().seq() % 256 != 0)
+    {
+        JLOG(j_.warn()) << "N-UNL: applyUNLModify, not a flag ledger, seq="
+                        << view().seq();
+        return tefFAILURE;
+    }
+
+    if (!ctx_.tx.isFieldPresent(sfUNLModifyDisabling) ||
+        ctx_.tx.getFieldU8(sfUNLModifyDisabling) > 1 ||
+        !ctx_.tx.isFieldPresent(sfLedgerSequence) ||
+        !ctx_.tx.isFieldPresent(sfUNLModifyValidator))
+    {
+        JLOG(j_.warn()) << "N-UNL: applyUNLModify, wrong Tx format.";
+        return tefFAILURE;
+    }
+
+    bool disabling = ctx_.tx.getFieldU8(sfUNLModifyDisabling);
+    auto seq = ctx_.tx.getFieldU32(sfLedgerSequence);
+    if (seq != view().seq())
+    {
+        JLOG(j_.warn()) << "N-UNL: applyUNLModify, wrong ledger seq=" << seq;
+        return tefFAILURE;
+    }
+
+    Blob validator = ctx_.tx.getFieldVL(sfUNLModifyValidator);
+    if (!publicKeyType(makeSlice(validator)))
+    {
+        JLOG(j_.warn()) << "N-UNL: applyUNLModify, bad validator key";
+        return tefFAILURE;
+    }
+
+    JLOG(j_.info()) << "N-UNL: applyUNLModify, disabling=" << disabling
+                    << " seq=" << seq
+                    << " validator data:" << strHex(validator);
+
+    auto const k = keylet::negativeUNL();
+    SLE::pointer nUnlObject = view().peek(k);
+    if (!nUnlObject)
+    {
+        nUnlObject = std::make_shared<SLE>(k);
+        view().insert(nUnlObject);
+    }
+
+    bool found = false;
+    if (nUnlObject->isFieldPresent(sfNegativeUNL))
+    {
+        auto const& nUnl = nUnlObject->getFieldArray(sfNegativeUNL);
+        for (auto it = nUnl.begin(); it != nUnl.end(); ++it)
+        {
+            if (it->isFieldPresent(sfPublicKey) &&
+                it->getFieldVL(sfPublicKey) == validator)
+                found = true;
```

Comment:
You could `break` here, no need to check the other elements once this is `true`. But I'd probably wrap this in a lambda instead and return (that way `found` could be `const`)
```

    bool const found = [&] {
        if (nUnlObject->isFieldPresent(sfNegativeUNL))
        {
            auto const& nUnl = nUnlObject->getFieldArray(sfNegativeUNL);
            for (auto it = nUnl.begin(); it != nUnl.end(); ++it)
            {
                if (it->isFieldPresent(sfPublicKey) &&
                    it->getFieldVL(sfPublicKey) == validator)
                    return true;
            }
        }
        return false;
    }();
```

Reply:
Right, changed, Thanks.